hazelcast/src/main/java/com/hazelcast/map/impl/recordstore/Storage.java

```diff
@@ -62,4 +63,7 @@
      * @return sampled entries.
      */
     Iterable<LazyEntryViewFromRecord> getRandomSamples(int sampleCount);
+
+    MapKeyIteratorResult fetchNext(int tableIndex, int size);
+
```

Comment:
This is a low level operation on the `Storage`. Using the name Iterator here is a bit confusing since it is neither an Iterator nor it doesn't know anything about the partitionIteration. Maybe just `MapKeys` or `KeySet` would sound better?